cfpb/consumerfinance.gov
# New models for Filing Instruction Guide content page

Previously, the filing instructions guide used the data model that we had originally designed for the Reports as Webpages project. We borrowed that code to get the nice design and layout with the Table of Contents for long, detailed content pages. However, the pages used a structured clusterable data model instead of a streamfield, and wouldn't allow wagtail content blocks (e.g. table, well) in the content. This PR switches the data models to a streamfield while trying to retain the sidenav TOC functionality.

If merged, will supersede #7054.

## Changes

- Totally gut and simplify the models in the `filing_instruction_guide` app to use a StreamField paradigm
  - This allows us to use simplified logic for generating templates and assigning section numbers to the sections in the page content
 - Rather than assigning section numbers at page load time, assign them at page save time (via a custom Wagtail form) instead

## How to test this PR

1. Pull the branch and run it
2. Create a FIG Content Page
3. Add sections, subsections, and level-three subsections to the page
4. Publish or preview the page
5. See that the sections are numbered in a logical way in both the side-nav and the page content. 
6. Check that the anchor links in the side-nav work to get you to the correct content section. Check that you can click a section header to anchor to that section.

## Screenshots


## Notes and todos

Todo:
- The TOC is supposed to become an accordion at smaller screen widths, but that's currently busted.
- I'd like to find a better way to display the section ID field in the Wagtail Admin to make it clear it's programmatically edited, not a field for the content editor to edit.

Note: Eventually, we'll make this page type (currently called FIG Content Page) into a more generic page type (probably Table of Contents Page) so it can be used for a variety of projects. For now, the code and models are still changing too much, and the styling isn't ready.


## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
